pkg/appgw/ingress_rules.go

```diff
@@ -14,39 +14,26 @@ import (
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/environment"
 )
 
-func (c *appGwConfigBuilder) getFrontendPortsFromIngress(ingress *v1beta1.Ingress, env environment.EnvVariables) map[Port]interface{} {
-	frontendPorts := make(map[Port]interface{})
-	for ruleIdx := range ingress.Spec.Rules {
-		rule := &ingress.Spec.Rules[ruleIdx]
-		if rule.HTTP == nil {
-			continue
-		}
-
-		ruleFrontendPorts, _ := c.processIngressRule(rule, ingress, env)
-		for port := range ruleFrontendPorts {
-			frontendPorts[port] = nil
-		}
-	}
-	return frontendPorts
-}
-
-func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, env environment.EnvVariables) map[listenerIdentifier]listenerAzConfig {
+func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, env environment.EnvVariables, additionalHostnames []string) map[listenerIdentifier]listenerAzConfig {
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
 	for ruleIdx := range ingress.Spec.Rules {
 		rule := &ingress.Spec.Rules[ruleIdx]
 		if rule.HTTP == nil {
 			continue
 		}
-
-		_, ruleListeners := c.processIngressRule(rule, ingress, env)
+		var addHNames []string
```

Comment:
Finding it hard to see if we really need to do this
Lets say Cx wants to apply annotation to all the path-rules in the ingress. This check will not allow them to do that and will generate 2 listener with hostnames for first path only ? Please correct me if I am wrong. 

I think we are making it more complex to use rather than documenting the failure case where customers have to split the ingress.